core/dvector.h

```diff
@@ -362,27 +347,26 @@ class PoolVector {
 	template <class MC>
 	void fill_with(const MC &p_mc) {
 
-		int c = p_mc.size();
-		resize(c);
+		resize(p_mc.size());
 		Write w = write();
 		int idx = 0;
-		for (const typename MC::Element *E = p_mc.front(); E; E = E->next()) {
 
+		for (const typename MC::Element *E = p_mc.front(); E; E = E->next()) {
 			w[idx++] = E->get();
 		}
 	}
 
 	void remove(int p_index) {
 
-		int s = size();
-		ERR_FAIL_INDEX(p_index, s);
+		int len = size();
+		ERR_FAIL_INDEX(p_index, len);
 		Write w = write();
-		for (int i = p_index; i < s - 1; i++) {
 
+		for (int i = p_index; i < len - 1; i++) {
 			w[i] = w[i + 1];
 		};
-		w = Write();
```

Comment:
Should not be just removed, this line drops `w`'s lock on the allocation. For consistency, could instead be replaced with a block controlling `w` lifetime that closes before the call to resize.